docs/testplan/GCU-Dynamic-ACL-testplan.md

```diff
@@ -0,0 +1,438 @@
+# Dynamic ACL Update via GCU Test Plan
+
+## Overview
+
+This test plan will certify that Generic Config Updater (GCU) is able to properly add, remove, and update ACL Table Types, ACL Tables, and ACL Rules.
+
+## Testbed
+
+The test will run on T0 testbeds.
+
+## Setup Configuration
+
+No setup pre-configuration is required, the test will configure and return the testbed to its original state.
+
+## Testing Plan
+
+To test the capability of GCU to dynamically update ACLs, we will utilize various Json Patch files to create, update, and remove various ACL Tables and Rules.  The contents of the Json Patch files, as well as additional details about verification processes, will be defined in the last section of this document, **JSON Patch Files and Expected Results**.
+
+### Test Case # 1 - Create and apply custom ACL table without rules
+
+#### Test Objective
+
+Verify that we can utilize GCU to create a custom ACL Table Type, and then create an ACL Table from this type
+
+#### Testing Steps
+
+- Create a new ACL Table Type
+
+- Create an ACL Table utilizing this ACL Table Type
+
+- Verify that both operations were successful
+
+- Verify that output of "show acl table {tablename}" matches expected output
+
+### Test Case # 2 - Create a drop rule within custom table
+
+#### Test Objective
+
+Verify that we can create a single drop rule utilizing GCU
+
+#### Testing Steps
+
+- Create a new ACL Table Type
```

Comment:
Please consider to make the steps as a fixture so tha we only need to create the ACL tables once and use is in all test cases. 
Example
```
def create_custom_acl_table:
    # Create a new ACL Table Type
    # Create an ACL Table utilizing this ACL Table Type
    yield
    # Remove ACL table
    # Remove ACL table type

```